### PR TITLE
[release-v0.49] Must clear "cloneRequest" annotation on restore of PVC 

### PR DESCRIPTION
**Pull request by brybacki**
**What this PR does / why we need it**:

This is manual cherry pick of https://github.com/kubevirt/kubevirt/pull/8078

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by brybacki** [comment]
I think I have found a problem, the CDI deployed for tests is 1.43.0 and the cdi fix landed on branch 1.43 but was not released, so needs a release and version bump to work in tests.

**Comment by brybacki** [comment]
This is the fix on the cdi side: https://github.com/kubevirt/containerized-data-importer/pull/2325

File: tests/storage/restore.go [hunk 1]
```diff
@@ -1083,6 +1089,120 @@ var _ = SIGDescribe("[Serial]VirtualMachineRestore Tests", func() {
...
+					tests.SetDataVolumePVCStorageClass(dataVolume, snapshotStorageClass)
+					tests.SetDataVolumePVCSize(dataVolume, "6Gi")
+					vmi := tests.NewRandomVMIWithDataVolume(dataVolume.Name)
+					tests.AddUserData(vmi, "cloud-init", tests.BashHelloScript)
+					vm := tests.NewRandomVirtualMachine(vmi, false)
+					tests.AddDataVolumeTemplate(vm, dataVolume)
+					return vm
+				}
+
+				It("should restore a vm that boots from a network cloned datavolumetemplate to the same VM", func() {
```
Comment (mhenriks) [inline tests/storage/restore.go:1179]: Am I correct in assuming that this is the test that fails?  As long as the following one passes, I am okay with deleting this one.  The code change is still exercised by following test.

Comment (brybacki) [reply tests/storage/restore.go:1179]: Yes, VM with datavolume works on older CDI, datavolume template only on the with the fix. 

And you are right, here we only care if the original problem was solved. Both tests pass on main, so I'll delete this one and rerun.

**Comment by brybacki** [comment]
@mhenriks all tests green, can you review and check if I did the manual cherry pick with merge conflicts correctly?

**Pull request merged**